scripts/s2i-generate.sh

```diff
@@ -31,14 +37,28 @@ phase "Inspecting context subdirectory '${PARAMS_SUBDIRECTORY}'"
 [[ ! -d "${PARAMS_SUBDIRECTORY}" ]] &&
     fail "Application source code directory not found at '${PARAMS_SUBDIRECTORY}'"
 
+phase "Preparing the environment variables"
+
+# create directory in the specified environment path only if its not already present
+if [[ ! -d "${ENV_DIR}" ]]; then
+    phase "Creating directory '${ENV_DIR}'"
+    mkdir -v -p "${ENV_DIR}"
+fi
+
+# add the environment variables that are sent as command line arguments from ENV_VARS parameter
+for env_var in "${env_vars[@]}"; do
+  echo $env_var >> /s2i-env/env
+done
+
 #
 # S2I Generate
 #
 
 phase "Generating the Containerfile for S2I builder image '${S2I_BUILDER_IMAGE}'"
 s2i --loglevel "${S2I_LOGLEVEL}" \
     build "${PARAMS_SUBDIRECTORY}" "${S2I_BUILDER_IMAGE}" \
-        --as-dockerfile "${S2I_CONTAINERFILE_PATH}"
+        --as-dockerfile "${S2I_CONTAINERFILE_PATH}" \
+        --environment-file /s2i-env/env
```

Comment:
Do we always create the environment file? Either the script needs to make sure the file is present, or we omit the argument when no environment variables are provided.

Also, please make sure to support the common mount path mentioned before 🙏